packages/stack/embarkjs/src/index.js

```diff
@@ -66,12 +67,15 @@ class EmbarkJS {
     if (moduleName === 'blockchain') moduleName = 'Blockchain';
     if (moduleName === 'names') moduleName = 'Names';
 
-    const registerProviderCode = `
-      const __embark_${stackName}_${pluginName} = require('${packageName}');
-      EmbarkJS.${moduleName}.registerProvider('${pluginName}', __embark_${stackName}_${pluginName}.default || __embark_${stackName}_${pluginName});
-    `;
+    if (moduleName != 'Blockchain' || this.blockchainConfig.enabled) {
```

Comment:
Not sure if this condition works. As you can see above on line 66, it seems like it should be `blockchain` (with a small B)
Also, would be better to use `!==`

Reply:
@jrainville the devil is in the details. Notice how `moduleName` gets mutated when it's `blockchain`. So this check is fine. Will switch to `!==`, thanks!

Reply:
Trying to parse this...
> If the module name is not Blockchain OR if the blockchain is enabled, then do...
From what follows inside the conditional the purpose of the conditional itself isn't too clear to me. Can you explain it, and do you think an explanatory comment in the source code might be helpful for future maintenance?

Reply:
Good point. So the code that gets generated in `registerProviderCode`, is generated for all the above mentioned stack modules (storage, blockchain, whisper etc).

Without this condition, this code gets generated and eval'd for all of them, but with this commit we're actually making it work that blockchain can be disabled, which means, that provider code should not be eval'd.

That's why the condition says: If it's not `Blockchain`, **or** if blockchain is enabled, generate and eval this code.

I should have definitely added a comment because the same check should be done for all the other stack components as well, but this PR only addresses Blockchain.